sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2189,7 +2194,7 @@ class Analyzer(override val catalogManager: CatalogManager)
 
     private def processV2ScalarFunction(
         scalarFunc: ScalarFunction[_],
-        inputType: StructType,
+        argumentTypes: StructType,
```

Comment:
do we still need this parameter? we can simplify the scala function branch
```
scalarFunc.getClass.getDeclaredMethod("produceResult", classOf[InternalRow]) ...
```

Reply:
Yep no longer needed. Will remove.